trlx/utils/logging.py

```diff
@@ -0,0 +1,352 @@
+
+
+def _reset_library_root_logger() -> None:
+    global _default_handler
+
+    with _lock:
+        if not _default_handler:
+            return
+
+        library_root_logger = _get_library_root_logger()
+        library_root_logger.removeHandler(_default_handler)
+        library_root_logger.setLevel(logging.NOTSET)
+        _default_handler = None
+
+
+def get_log_levels_dict():
+    return log_levels
+
+
+class MultiProcessAdapter(logging.LoggerAdapter):
+    """A logger adapter for handling multi-process logging"""
+
+    def log(self, level, msg, *args, **kwargs):
+        """
+        Consumes an additional kwarg called `ranks` to determine which processes should log.
+        NOTE: To specify all processes, pass in an empty list `ranks=[]`
+
+        Default: ["0"], i.e. only the main process logs
+        """
+        # By default, silence all non-main processes
+        ranks = kwargs.pop("ranks", ["0"])
+        should_log = os.environ.get("RANK", "0") in ranks or len(ranks) == 0
+        if self.isEnabledFor(level) and should_log:
+            msg, kwargs = self.process(msg, kwargs)
+            self.logger._log(level, msg, args, **kwargs)
+
+    def process(self, msg, kwargs):
+        this_rank = torch.distributed.get_rank() if torch.distributed.is_initialized() else 0
+        return f"[RANK {this_rank}] {msg}", kwargs
+
+
+def get_logger(name: Optional[str] = None) -> MultiProcessAdapter:
+    """
+    Returns a `logging.Logger` for `name` that can handle multiple processes
+
+    Args:
+        name: Name of the logger
+
+    Usage:
+        >> logger = get_logger(__name__)
+        >> logger.debug("Check the...", ranks=["0", "1"])  # Only main and rank 1 log
+    """
+    if name is None:
+        name = _get_library_name()
+    _configure_library_root_logger()
+    logger = logging.getLogger(name)
+    return MultiProcessAdapter(logger, {})
+
+
+def get_verbosity() -> int:
+    """
+    Return the current level for trlx's root logger as an int.
+    Returns:
+        `int`: The logging level.
+    <Tip>
+    trlx has following logging levels:
+    - 50: `trlx.logging.CRITICAL` or `trlx.logging.FATAL`
+    - 40: `trlx.logging.ERROR`
+    - 30: `trlx.logging.WARNING` or `trlx.logging.WARN`
+    - 20: `trlx.logging.INFO`
+    - 10: `trlx.logging.DEBUG`
+    </Tip>"""
+
+    _configure_library_root_logger()
+    return _get_library_root_logger().getEffectiveLevel()
+
+
+def set_verbosity(verbosity: int) -> None:
+    """
+    Set the verbosity level for the trlX's root logger.
+    Args:
+        verbosity (`int`):
+            Logging level, e.g., one of:
+            - `trlx.logging.CRITICAL` or `trlx.logging.FATAL`
+            - `trlx.logging.ERROR`
+            - `trlx.logging.WARNING`
+            - `trlx.logging.INFO`
+            - `trlx.logging.DEBUG`
+    """
+
+    _configure_library_root_logger()
+    _get_library_root_logger().setLevel(verbosity)
+
+
+def set_verbosity_info():
```

Comment:
so much code that does the same thing ... just give one option like the python library 

```python
trlx.logging.set_level(trlx.logging.INFO)
```